### PR TITLE
opensubdiv, enable secondary architecture, fix install paths

### PR DESCRIPTION
**Pull request by Begasus**

### CODE REVIEW

**Comment by Begasus** [comment]
@extrowerk can you check on x86_64?
@X547 could you check the build on 32bit too and see if blender can find it?

File: media-libs/opensubdiv/opensubdiv-3.4.0.recipe [hunk 1]
```diff
@@ -92,35 +86,35 @@ defineDebugInfoPackage opensubdiv$secondaryArchSuffix \
...
 	prepareInstalledDevelLibs \
 		libosdCPU \
 		libosdGPU
-
 	fixPkgconfig
 
-	packageEntries tools $binDir
-	packageEntries devel $developDir
+	packageEntries tools \
+		$prefix/bin
```
Comment (extrowerk) [inline media-libs/opensubdiv/opensubdiv-3.4.0.recipe]: I can't seem to find any PROVIDES_tools section....

File: media-libs/opensubdiv/patches/opensubdiv-3.4.0.patchset [hunk 2]
```diff
@@ -1,44 +0,0 @@
...
-index 3920896..254a4db 100644
---- a/CMakeLists.txt
-+++ b/CMakeLists.txt
-@@ -337,7 +337,7 @@ endif()
- if(NOT NO_CUDA)
-     find_package(CUDA 4.0)
- endif()
--if(NOT NO_GLFW AND NOT NO_OPENGL AND NOT ANDROID AND NOT IOS)
-+if(NOT NO_GLFW AND NOT NO_OPENGL AND NOT ANDROID AND NOT IOS AND NOT HAIKU)
-     find_package(GLFW 3.0.0)
```
Comment (extrowerk) [inline media-libs/opensubdiv/patches/opensubdiv-3.4.0.patchset]: GLFW is definetely not available on Haiku...

File: media-libs/opensubdiv/opensubdiv-3.4.0.recipe [hunk 3]
```diff
@@ -92,35 +86,35 @@ defineDebugInfoPackage opensubdiv$secondaryArchSuffix \
...
-	cmake .. \
+	cmake .. -DCMAKE_BUILD_TYPE=Release \
+		-DCMAKE_INSTALL_PREFIX=$prefix \
+		-DCMAKE_BINDIR_BASE=$prefix/bin \
+		-DCMAKE_INCDIR_BASE=$includeDir \
+		-DCMAKE_LIBDIR_BASE=$libDir \
 		-DNO_EXAMPLES=ON \
 		-DNO_TUTORIALS=ON \
-		$cmakeDirArgs
+		-DNO_GLFW_X11=ON \
```
Comment (extrowerk) [inline media-libs/opensubdiv/opensubdiv-3.4.0.recipe]: Oh, i see.

**Comment by X547** [comment]
32 bit build is confirmed.

**Comment by X547** [comment]
Headers install location is wrong. Headers should be installed to `opensubdiv` folder.

File: media-libs/opensubdiv/patches/opensubdiv-3.4.0.patchset (hunk 2 above)
Comment (Begasus) [reply media-libs/opensubdiv/patches/opensubdiv-3.4.0.patchset]: cmake checks if it's available, if not it sets a flag for it, doesn't bail out on it

File: media-libs/opensubdiv/opensubdiv-3.4.0.recipe (hunk 3 above)
Comment (Begasus) [reply media-libs/opensubdiv/opensubdiv-3.4.0.recipe]: It actually failed on not finding X11 for this

**Comment by Begasus** [comment]
> Headers install location is wrong. Headers should be installed to `opensubdiv` folder.

Then it should use "-DCMAKE_INCDIR_BASE=$includeDir/opensubdiv" then? (need to check)

**Comment by X547** [comment]
> Then it should use "-DCMAKE_INCDIR_BASE=$includeDir/opensubdiv"? (need to check)
I already made this change locally and managed to build Blender on 32 bit with opensubdiv.

**Comment by Begasus** [comment]
Pushed the change for the headers installation path, didn't check yet, but could you confirm @X547 ?

File: media-libs/opensubdiv/opensubdiv-3.4.0.recipe [hunk 4]
```diff
@@ -7,26 +7,20 @@ surface matches Pixar's Renderman to numerical precision."
...
 libVersion="$portVersion"
 libVersionCompat="$libVersion compat >= ${libVersion%%.*}"
 
 PROVIDES="
 	opensubdiv$secondaryArchSuffix = $portVersion
-	cmd:far_perf = $portVersion
-	cmd:far_regression = $portVersion
-	cmd:hbr_baseline = $portVersion
-	cmd:hbr_regression = $portVersion
-	cmd:stringify = $portVersion
```
Comment (X547) [inline media-libs/opensubdiv/opensubdiv-3.4.0.recipe]: Why remove commands?

Comment (Begasus) [reply media-libs/opensubdiv/opensubdiv-3.4.0.recipe]: They are provided in the _tools package (and not by the base package)

Comment (extrowerk) [reply media-libs/opensubdiv/opensubdiv-3.4.0.recipe]: They moved to _tool subpackage, but i can't seems to find the PROVIDES section for the subpackage.

Comment (Begasus) [reply media-libs/opensubdiv/opensubdiv-3.4.0.recipe]: Here? https://github.com/haikuports/haikuports/blob/ee0725aae337f986e134c26dea661c91d3cd4708/media-libs/opensubdiv/opensubdiv-3.4.0.recipe#L41

Comment (extrowerk) [reply media-libs/opensubdiv/opensubdiv-3.4.0.recipe]: I see, but why can't we see that in the diff? Or what the heck is going on here?

Comment (Begasus) [reply media-libs/opensubdiv/opensubdiv-3.4.0.recipe]: You already added it to the previous version, so no need to change anything there :)

**Comment by X547** [comment]
32 build confirmed, paths looks OK.

**Comment by Begasus** [comment]
> 32 build confirmed, paths looks OK.

Yeah, but seems Blender is still not picking it up (looking at cmake output), maybe it needs "OPENSUBDIV_INCLUDE_DIR" defined?

**Comment by X547** [comment]
I already uploaded new blender patch.

**Comment by Begasus** [comment]
> I already uploaded new blender patch.

OK, checked the changes for the recipe, not the patchset, re-check here :)

**Comment by Begasus** [comment]
Any changes needed here? LGTM

**Comment by X547** [comment]
Looks OK for me.

**Comment by Begasus** [comment]
Thanks for checking up :+1:

**Pull request merged**